### PR TITLE
Exclude project directory from caching source space

### PR DESCRIPTION
**Pull request by furushchev**
Hope this fix https://app.asana.com/0/1112383576851259/1203393813469035

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -184,6 +185,19 @@ jobs:
...
+        with:
+          method: load
+          key: setup_catkin_workspace_source_space
+          path: ./source_space
+          remote_path: s3://${{ secrets.AWS_S3_BUCKET_NAME }}
+      - name: Check setup_catkin_workspace cached source space
+        shell: bash
+        run: |
+          cd source_space
+          test -e geometry2
```
Comment (garaemon) [inline .github/workflows/test.yml:199]: Why check geometry2? please leave comment

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -184,6 +185,19 @@ jobs:
...
+          test ! -e project/package.xml
```
Comment (garaemon) [inline .github/workflows/test.yml:200]: Why is this line required? Please leave comment

File: setup_catkin_workspace/action.yml [hunk 3]
```diff
@@ -75,3 +75,4 @@ runs:
         key: ${{ inputs.cache_source_space_id }}
         path: ${{ inputs.catkin_workspace }}/src
         remote_path: ${{ inputs.cache_remote_path }}
+        exclude: 'project/**'
```
Comment (garaemon) [inline setup_catkin_workspace/action.yml:78]: Why does this line exclude `project/**`? Please leave comment.

**Review by garaemon — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (furushchev) [reply .github/workflows/test.yml:201]: In the workflow, `ros/geometry_tutorials` package is used as the tested repository and `geometry2` as its dependency, and the step is to ensure the dependency in the source space is cached.
Added a comment.

File: .github/workflows/test.yml (hunk 2 above)
Comment (furushchev) [reply .github/workflows/test.yml:202]: This is to ensure the project directory is excluded from the cache.

File: setup_catkin_workspace/action.yml (hunk 3 above)
Comment (furushchev) [reply setup_catkin_workspace/action.yml]: Substituted with `project_name` parameter and added the description.

Comment (furushchev) [reply setup_catkin_workspace/action.yml]: Created related PR to support this feature: https://github.com/GITAI/gitai_ci/pull/169
